### DISCUSSION TITLE
vaultwarden debian

### DISCUSSION THREAD

**Discussion by frezeen** [tteck/Proxmox#2547] [General]
lxc vaultwarden script stop with errors
Failure to comply with the guidelines

**Comment by tteck** [comment]
You stated in https://github.com/tteck/Proxmox/issues/2546
> first try with default setting and waltwarden.bin compiling stopped with out of memory error on proxmox syslog, try with custom settings , 8gb ram lxc, and creation si ended without error.
i think, but i cant confirm this, 4gb ram is low for to complete lxc creation.

I consider that a suggestion to increase the default build RAM settings. So yes that's a "Failure to comply with the guidelines"

**Comment by frezeen** [comment]
what i want to said is:
your vaultwarden debian script has 2 phases. in phase 1 u set 4gb ram for to compile and install vaultwarden , then phase 2, at end, set the CT with standard 512mb ram setting. ok, script as it is dont complete and stuck in compiling last big file.  u wait for infinite time for nothing seeing the processing that look going without errors (i wait 45 mins before i understand something is going wrong and stop it) or u go to proxmox syslog and find the memory error, 4gb ram isnt enough. isnt a suggestion, u need to set more than 4gb for to fix the script installation infinite stuck . for me, the place for this is issue section and need to be fixed but u are the boss. sorry

**Reply by tteck** [reply]
I have no issues with 4GB RAM
```
 _    __            ____ _       __               __         
| |  / /___ ___  __/ / /| |     / /___ __________/ /__  ____ 
| | / / __ `/ / / / / __/ | /| / / __ `/ ___/ __  / _ \/ __ \
| |/ / /_/ / /_/ / / /_ | |/ |/ / /_/ / /  / /_/ /  __/ / / /
|___/\__,_/\__,_/_/\__/ |__/|__/\__,_/_/   \__,_/\___/_/ /_/ 
                                                             
Using Advanced Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 1
Using Root Password: Automatic Login
Container ID: 110
Using Hostname: vaultwarden
Using Disk Size: 6
Allocated Cores: 4
Allocated RAM: 4096
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Using APT-Cacher IP Address: Default
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using MAC Address: AE:1A:60:07:B7:59
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a Vaultwarden LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using thin-pool for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 110 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.250 
 ✓ IPv4 Internet Connected
 ✗ IPv6 Internet Not Connected
 ✓ DNS Resolved github.com to 140.82.114.4
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Installed Rust
 ✓ Built Vaultwarden 1.30.3
 ✓ Downloaded Web-Vault v2024.1.2b
 ✓ Created Service
 ✓ Customized Container
 ✓ Cleaned
 ✓ Set Container to Normal Resources
 ✓ Completed Successfully!

Vaultwarden should be reachable by going to the following URL.
         http://192.168.86.250:8000 
```
So, you're suggesting I should make adjustments tailored to your particular system?

**Comment by frezeen** [comment]
this is really strange... i test soem mins ago..
my system has 64gb ram and 32gb free on proxmox host.
when i try lxc build with default , compiling stuck on vaulwardden.bin  process without errors about 10 mins later, cpu% go down to 1% and i got this in syslog and i need to ctrl+c on going build process in lxc shell for to exit without errors writed there 

`Feb 23 14:00:46 Proxmox kernel: cargo invoked oom-killer: gfp_mask=0x100cca(GFP_HIGHUSER_MOVABLE), order=0, oom_score_adj=0
Feb 23 14:00:46 Proxmox kernel: CPU: 1 PID: 255522 Comm: cargo Tainted: P           O       6.5.11-8-pve #1
.....
Feb 23 14:00:46 Proxmox kernel: memory: usage 4194304kB, limit 4194304kB, failcnt 25617772
Feb 23 14:00:46 Proxmox kernel: swap: usage 0kB, limit 524288kB, failcnt 0
.....
Feb 23 14:00:46 Proxmox kernel: oom-kill:constraint=CONSTRAINT_MEMCG,nodemask=(null),cpuset=ns,mems_allowed=0,oom_memcg=/lxc/110,task_memcg=/lxc/110/ns/system.slice/system-container\x2dgetty.slice/container-getty@1.service,task=rustc,pid=259686,uid=100000
Feb 23 14:00:46 Proxmox kernel: Memory cgroup out of memory: Killed process 259686 (rustc) total-vm:8054296kB, anon-rss:4114948kB, file-rss:0kB, shmem-rss:0kB, UID:100000 pgtables:14316kB oom_score_adj:0`

the thing only i see different from you is that i use a zfs-storage for my lxc creation... maybe lxc need more ram for to manage zfs storage?
